DCGAN/readme.md

```diff
@@ -0,0 +1,144 @@
+
+## Generative Adversarial Networks
+GANs are generally made up of two models: The Artist(Generator) and The Critic(Discriminator). The generator creates an image from random noise, and the discriminator evaluates the generated image from the images in the given dataset. We train the models by minmaxing the costs of the models. The generator tries to fool the discriminator by producing realistic looking images, and the discriminator becomes better in understanding the difference between real and fake images. This two player game improves the model until the generator produces images realistic images or the system reaches nash equilibrium.
+
+## Contents
+1. [Setup instructions and dependencies](#1-setup-instructions-and-dependencies)
+2. [Training model from scratch](#2-training-model-from-scratch)
+3. [Generating images from trained models](#3-generating-images-from-trained-models)
+4. [Model architecture](#4-model-architecture)
+5. [Repository overview](#5-repository-overview)
+6. [Results obtained](#6-results-obtained)
+7. [Observations](#7-observations)
+8. [Credits](#8-credits)
+
+## 1. Setup instructions and dependencies
+You may setup the repository on your local machine by either downloading it or running the following line on `cmd prompt`.
+
+``` Batchfile
+git clone https://github.com/Atharv24/GenZoo.git
+```
+
+The trained models are large in size and hence their Google Drive links are provided in the `model.txt` file.
+
+The data required for training is automatically downloaded when running `train.py`.
+
+All dependencies required by this repo can be downloaded by creating a virtual environment and running
+
+``` Batchfile
+pip install -r requirements.txt
+```
+
+## 2. Training model from scratch
+To train your own model from scratch, run
+
+```Batchfile
+python train.py -config path/to/config.ini
+```
+
++ The parameters for your experiment should be set in your `config.ini` file.
++ The training script will create a folder **exp_name** as specified in your `config.ini` file.
++ This folder will contain all data related to your experiment such as tensorboard logs, images generated during training and training checkpoints.
+
+## 3. Generating images from trained models
+To generated images from trained models, run
+
+```Batchfile
+python generate.py -dataset mnist/cifar -load_path path/to/directory/containing/checkpoint -grid_size n -save_path directory/where/images/are/saved
+```
+
+The arguments used are explained as follows
+
++ `-dataset` requires either `mnist` or `cifar` according to what dataset the model was trained on.
++ `-load_path` requires the path to directory containing the training checkpoints. The latest checkpoint is loaded from the directory.
++ `-grid_size` requires integer `n` and will generate n*n images in a grid.
++ `-save_path` requires the path to directory where the generated images will be saved. If the directory doesn't exist the script will create it.
```

Comment:
1. requires the path to `the` directory where ...
2. doesn't exist`,` the script will create it.